CMakeLists.txt

```
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.22)
+
+include(FetchContent)
+
+project(ClpIrV1Decoder
+    LANGUAGES CXX C
+)
+
+# Enforce "Unix Makefiles" because "Ninja" does not work with the Emscripten toolchain on CXX_20 sources.
+if(NOT CMAKE_GENERATOR MATCHES "Unix Makefiles")
+    message(FATAL_ERROR "This project is intended to be built with Unix Makefiles on a Unix system.")
+endif()
+
+# Enable compile commands by default if the generator supports it.
+if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE BOOL
+        "Enable/Disable output of compile commands during generation." FORCE)
+endif()
+
+# Set the default build type to Release if not specified
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build (Debug, Release, RelWithDebInfo, MinSizeRel)." FORCE)
+endif()
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0" CACHE STRING "Version of Boost library to be used when compiling CLP sources.")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+set(BOOST_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/build/wasm/boost" CACHE STRING
+    "Directory containing Boost sources, fetched from official release if non existent.")
+if (NOT EXISTS ${BOOST_SRC_DIR})
+    FetchContent_Declare(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        SOURCE_DIR ${BOOST_SRC_DIR}
+    )
+    message(STATUS "Fetching Boost from official release.")
+    FetchContent_MakeAvailable(Boost)
+endif()
+
+set(BIN_NAME "ClpIrV1Decoder" CACHE STRING "Binary name for the generated .js file and .wasm file of CLP IR V1 Decoder.")
+add_executable(${BIN_NAME})
+
+target_compile_features(${BIN_NAME}
+    PRIVATE
+    cxx_std_20
+)
+
+set(CMAKE_EXECUTABLE_SUFFIX ".js")
+
+target_link_options(${BIN_NAME} PRIVATE
+    "-s EXPORTED_FUNCTIONS=['_malloc','_free']"
+    "-s MODULARIZE=1"
+    "-s EXPORT_ES6=1"
+    "-s ENVIRONMENT=worker"
+    "-lembind"
+    "-s ALLOW_MEMORY_GROWTH=1"
+    "-s WASM_BIGINT"
+    "-flto"
+    "--closure 1"
+)
+
+target_compile_definitions(${BIN_NAME} PUBLIC SPDLOG_FMT_EXTERNAL=1)
+
+target_include_directories(
+        ${BIN_NAME}
+        PRIVATE
+        ${BOOST_SRC_DIR}
+        src/
+        src/submodules/clp/components/core/src
+        src/submodules/clp/components/core/src/clp
+        src/submodules/clp/components/core/submodules
+        src/submodules/fmt/include
+        src/submodules/spdlog/include
+        src/submodules/zstd/lib
+)
```

nit:
1. Let's put the target name on the same line of the command.
2. Let's be consistent wit the indentations (4 spaces).
```suggestion
target_include_directories(${BIN_NAME}
    PRIVATE
    ${BOOST_SRC_DIR}
    src/
    src/submodules/clp/components/core/src
    src/submodules/clp/components/core/src/clp
    src/submodules/clp/components/core/submodules
    src/submodules/fmt/include
    src/submodules/spdlog/include
    src/submodules/zstd/lib
)
```